app/models/file_depot_swift.rb

```diff
@@ -0,0 +1,66 @@
+class FileDepotSwift < FileDepot
+  def self.uri_prefix
+    "swift"
+  end
+
+  def self.validate_settings(settings)
+    new(:uri => settings[:uri]).verify_credentials(nil, settings.slice(:username, :password))
+  end
+
+  def connect(options = {})
+    openstack_handle(options).connect(options)
+  end
+
+  def openstack_handle(options = {})
+    require 'manageiq/providers/openstack/legacy/openstack_handle'
+    @openstack_handle ||= begin
+      username = options[:username] || authentication_userid(options[:auth_type])
+      password = options[:password] || authentication_password(options[:auth_type])
+      uri      = options[:uri]
+      address  = URI(uri).host
+      port     = URI(uri).port
+
+      extra_options = {
+        :ssl_ca_file    => ::Settings.ssl.ssl_ca_file,
+        :ssl_ca_path    => ::Settings.ssl.ssl_ca_path,
+        :ssl_cert_store => OpenSSL::X509::Store.new
+      }
+      extra_options[:domain_id]         = v3_domain_ident
+      extra_options[:service]           = "Compute"
+      extra_options[:omit_default_port] = ::Settings.ems.ems_openstack.excon.omit_default_port
+      extra_options[:read_timeout]      = ::Settings.ems.ems_openstack.excon.read_timeout
+      begin
+        OpenstackHandle::Handle.new(username, password, address, port, keystone_api_version, security_protocol, extra_options)
+      rescue => err
+        logger.error("Error connecting to Swift host #{address}. #{err}")
+        msg = "Error connecting to Swift host #{address}. #{err}"
```

Comment:
Can we deduplicate the strings here by assigning to `msg` first?

Reply:
Done in three instances in this file.